### ISSUE TITLE
Dota 2 stutters specifically on my account.

### ISSUE THREAD

**Issue by Endeavour18** [ValveSoftware/Dota2-Gameplay#21922]
### Description

Basically, my account has some kind of problem that causes the game to lag on any PC, not just mine. I asked a friend to test it on his computer, and he also experienced lag using my account. Additionally, he let me borrow his account to test on my PC, and his account ran just fine. We played turbo matches just to test out but it happens when I play normal games too.

The 'stuttering' I’m talking about happens when the game runs at 140+ FPS but feels like I’m seeing below 50 FPS (that does not happen on my friend's account tho). It’s very awkward, and the lag gets much worse during teamfights or engages. I also noticed the stuttering occurs when I’m not pressing buttons or when I’m dead, but I could be wrong.

I’ve always had this issue, but it became even worse since the Crownfall update. I can play any AAA game without a problem on this system.

my ID: 119608754

**What I have tried:**

**PC related:**
Reinstalled OS and ran Dota only with the necessary programs/drivers. Installed discord, etc. later;
Cleaned up PC and changed thermal paste;
Undervolted / Overclocked CPU/GPU;
Bought a new PSU in case of low power supply;
Bought a new high quality DP cable;
Disabled/enabled FreeSync from my monitor/driver configs;
Disabled trusted computing from BIOS;
Tried different GPU driver versions (most recent version and older versions).
Used different benchmark tools to test out the hardware. Such as Userbenchmark, 3Dmark and Black Myth Wukong Benchmark, everything on ULTRA. Also did monitoring on temperatures, etc. while playing dota.
Turned off windows defender;
Turned off my 2nd monitor.

**UserBenchmarks:**
![image](https://github.com/user-attachments/assets/37612e84-39da-47f3-98dd-f5e0d77b4653)
![image](https://github.com/user-attachments/assets/9c51f519-42df-4937-9677-bb3c0d95b715)
---
**Dota related:**
Tried Vulkan and DX11 with various configurations on my GPU driver/dota;
Disabled Steam Cloud Save for Dota and any other game.
Removed everything from Profile + miniprofile to make it "clean";
Removed skins (towers, weather effects, hero skins, couriers, wards, EVERYTHING);
Deleted over 100 friends;
Deleted over 1000 items;
Lowered the graphics to the minimum possible;
Reinstalled the game and checked files integrity;
Moved from NVME to SSD and vice versa;
Made a fresh userconfig file deleting my 570 folder;
Copied my friend's 570 config folder and pasted to mine;
Tried various launch options such as "high", "threads", "mainthreadpriority", "novid", and so on.

**[Video playing on my main account with stutter.](https://youtu.be/C8pTgslFBuU)**

Teamfight timestamps: [10:10](https://youtu.be/C8pTgslFBuU?t=610), [14:10](https://youtu.be/C8pTgslFBuU?t=851), [14:46](https://youtu.be/C8pTgslFBuU?t=886), [19:00](https://youtu.be/C8pTgslFBuU?t=1140), 

Update: I recorded the best example possible. Stuttering occurs from the start to the finish of this video. Notice that the FPS is high, yet the game looks very bad.
https://youtu.be/HA2CDn-uTu4
 
**[Video playing on my friend's account without the stutter](https://youtu.be/V3tvWP1d0Xo)**

Teamfight timestamps: [11:04](https://youtu.be/V3tvWP1d0Xo?t=663), [12:18](https://youtu.be/V3tvWP1d0Xo?t=738), [13:06](https://youtu.be/V3tvWP1d0Xo?t=786), [14:50](https://youtu.be/V3tvWP1d0Xo?t=890), [16:39](https://youtu.be/V3tvWP1d0Xo?t=998), [20:50](https://youtu.be/V3tvWP1d0Xo?t=1245)

**My system information (from Steam).**

**Computer Information:**
Manufacturer: ASUS
Model: System Product Name
Form Factor: Desktop
No Touch Input Detected
Processor Information:
CPU Vendor: AuthenticAMD
CPU Brand: AMD Ryzen 7 5700X 8-Core Processor
CPU Family: 0x19
CPU Model: 0x21
CPU Stepping: 0x2
CPU Type: 0x0
Speed: 3400 MHz
16 logical processors
8 physical processors
Hyper-threading: Supported
FCMOV: Supported
SSE2: Supported
SSE3: Supported
SSSE3: Supported
SSE4a: Supported
SSE41: Supported
SSE42: Supported
AES: Supported
AVX: Supported
AVX2: Supported
AVX512F: Unsupported
AVX512PF: Unsupported
AVX512ER: Unsupported
AVX512CD: Unsupported
AVX512VNNI: Unsupported
SHA: Supported
CMPXCHG16B: Supported
LAHF/SAHF: Supported
PrefetchW: Unsupported
Operating System Version:
Windows 10 (64 bit)
NTFS: Supported
Crypto Provider Codes: Supported 311 0x0 0x0 0x0
Video Card:
Driver: AMD Radeon RX 6750 XT
DirectX Driver Name: atidx9loader32.dll
Driver Version: 32.0.11037.4004
DirectX Driver Version: 32.0.11037.4004
Driver Date: 8 23 2024
OpenGL Version: 4.6
Desktop Color Depth: 32 bits per pixel
Monitor Refresh Rate: 60 Hz
DirectX Card: AMD Radeon RX 6750 XT
VendorID: 0x1002
DeviceID: 0x73df
Revision: 0xc0
Number of Monitors: 2
Number of Logical Video Cards: 2
No SLI or Crossfire Detected
Primary Display Resolution: 2560 x 1440
Desktop Resolution: 4480 x 1440 (Dual monitor)
Primary Display Size: 23.54" x 13.23" (26.97" diag), 59.8cm x 33.6cm (68.5cm diag)
Primary Bus Type Not Detected
Primary VRAM Not Detected
Supported MSAA Modes: 2x 4x 8x
Sound card:
Audio device: SteelSeries Sonar - Media (Stee
Memory:
RAM: 32678 Mb
VR Hardware:
VR Headset: None detected
Miscellaneous:
UI Language: English
Media Type: Undetermined
Total Hard Disk Space Available: 571641 MB
Largest Free Hard Disk Block: 138874 MB
OS Install Date: ago 30 2024
Game Controller: None detected
MAC Address hash: 06b7279ea6d98e5e22689878e6bcad8b4c94b555
Storage:
Disk serial number hash: e021ac99
Number of SSDs: 3
SSD sizes: 240G,240G,120G
Number of HDDs: 0
Number of removable drives: 0

====================

**My friend's system information**

Computer Information:
Manufacturer: Gigabyte Technology Co., Ltd.
Model: B360M-D3H
Form Factor: Desktop
No Touch Input Detected
Processor Information:
CPU Vendor: GenuineIntel
CPU Brand: Intel(R) Core(TM) i5-8400 CPU @ 2.80GHz
CPU Family: 0x6
CPU Model: 0x9e
CPU Stepping: 0xa
CPU Type: 0x0
Speed: 2808 MHz
6 logical processors
6 physical processors
Hyper-threading: Unsupported
FCMOV: Supported
SSE2: Supported
SSE3: Supported
SSSE3: Supported
SSE4a: Unsupported
SSE41: Supported
SSE42: Supported
AES: Supported
AVX: Supported
AVX2: Supported
AVX512F: Unsupported
AVX512PF: Unsupported
AVX512ER: Unsupported
AVX512CD: Unsupported
AVX512VNNI: Unsupported
SHA: Unsupported
CMPXCHG16B: Supported
LAHF/SAHF: Supported
PrefetchW: Unsupported
Operating System Version:
Windows 10 (64 bit)
NTFS: Supported
Crypto Provider Codes: Supported 311 0x0 0x0 0x0
Video Card:
Driver: NVIDIA GeForce GTX 1060 6GB
DirectX Driver Name: nvldumd.dll
Driver Version: 31.0.15.5123
DirectX Driver Version: 31.0.15.5123
Driver Date: 1 18 2024
OpenGL Version: 4.6
Desktop Color Depth: 32 bits per pixel
Monitor Refresh Rate: 120 Hz
DirectX Card: NVIDIA GeForce GTX 1060 6GB
VendorID: 0x10de
DeviceID: 0x1c06
Revision: 0xa1
Number of Monitors: 2
Number of Logical Video Cards: 2
No SLI or Crossfire Detected
Primary Display Resolution: 1920 x 1080
Desktop Resolution: 3520 x 1080
Primary Display Size: 20.71" x 11.65" (23.74" diag), 52.6cm x 29.6cm (60.3cm diag)
Primary Bus: PCI Express 16x
Primary VRAM: 6143 MB
Supported MSAA Modes: 2x 4x 8x
Sound card:
Audio device: Alto-falantes (USB Audio Device
Memory:
RAM: 16317 Mb
VR Hardware:
VR Headset: None detected
Miscellaneous:
UI Language: English
Media Type: Undetermined
Total Hard Disk Space Available: 1473028 MB
Largest Free Hard Disk Block: 683869 MB
OS Install Date: fev 04 2024
Game Controller: None detected
MAC Address hash: 9d85c37950a4a2521baca1f8bb7d63d400535850
Storage:
Disk serial number hash: 0
Number of SSDs: 2
SSD sizes: 480G,128G
Number of HDDs: 2
HDD sizes: 1000G,15G
Number of removable drives: 0

### Example Match ID (and possibly Timestamp)

7952287517, 7955859875

**Comment by Endeavour18** [comment]
Update: I recorded the best example possible. Stuttering occurs from the start to the finish of this video. Notice that the FPS is high, yet the game looks very bad.
https://youtu.be/HA2CDn-uTu4

**Comment by IgorHoruzha** [comment]
Hello, to be honest, the issue isn't clearly visible in the video, but I believe I understand what the problem might be. There are two theories about it.

Could You please try the following instructions to test the first theory?
Following these steps will reset Your Dota 2 cloud settings and start with fresh configurations:

![image](https://github.com/user-attachments/assets/e40aa09e-ad5a-4bcf-b476-c0d7523d27ff)
![image](https://github.com/user-attachments/assets/06466eb5-5e24-4020-a454-4a3011f1b49a)

2. Reset Local Game Settings
Move this folder (don't delete it, please) to a different location:
C:\Program Files (x86)\Steam\userdata\[your SteamID]\570
![image](https://github.com/user-attachments/assets/c4383af1-b57a-43b8-8c97-43760bf42af4)


![image](https://github.com/user-attachments/assets/da45aa7f-9538-4ade-a68e-3007e1c68c21)

![image](https://github.com/user-attachments/assets/6c66c196-2e3b-4214-abec-2b442180eff9)

**Comment by Endeavour18** [comment]
@IgorHoruzha I already did that as mentioned in the issue. Check the windranger video, the lag occours more often on that one. Timestamp: https://youtu.be/HA2CDn-uTu4?t=586

**Comment by IgorHoruzha** [comment]
>Disabled Steam Cloud Save for Dota and any other game.
Removed everything from Profile + miniprofile to make it "clean";
Made a fresh userconfig file deleting my 570 folder;

I see Your point; it’s important to follow these steps in the correct sequence to avoid loading files from the cloud.
Have You also tried moving the following folder?
C:\Program Files (x86)\Steam\steamapps\common\dota 2 beta\game\dota\cfg

Could You also try using the -autoconfig launch option, please?

**Comment by Endeavour18** [comment]
@IgorHoruzha The sequence was right because all my hotkeys and configs got reseted. I also tried the autoconfig launch options and no sucess

**Comment by Endeavour18** [comment]
@IgorHoruzha That was one of the firsts tests I did before opening my case and cleaning up my PC.

**Comment by Endeavour18** [comment]
@IgorHoruzha and even if that was the case, why any other account run fine and not mine with the exact same settings? makes no sense

**Comment by IgorHoruzha** [comment]
>and even if that was the case, why any other account run fine and not mine with the exact same settings? makes no sense

I see Your point; it's possible that something is missing or that random external factors are affecting the statistics. Could You try turning off Microsoft Defender's real-time protection and test again?



I tried comparing the videos but still couldn't clearly identify the issue. I used Microsoft Clipchamp. Could You try the same method and focus on the moment the issue occurs?

Here are the links to the videos:

https://github.com/user-attachments/assets/cf133ecc-373a-4bbe-8e17-d804abca268d

https://github.com/user-attachments/assets/82cb67ea-3789-416e-b097-3bd2bf7b6e7e

Both videos You sent have a frame rate of 30 fps, with 1 frame every 33 ms


Additionally, could You check using the [UFO test](https://www.testufo.com/framerates-versus), please?

**Comment by Endeavour18** [comment]
@IgorHoruzha my windows defender (real time protection) is already disabled and the videos are in 2k resolution at 60fps

**Comment by IgorHoruzha** [comment]
Interesting, after downloading from YouTube, the frame rate of the video is 30 fps.

![image](https://github.com/user-attachments/assets/7855f3d2-6eec-4090-8cd2-2df70ab280cc)

Do You notice if one of the UFOs behaves similarly? Could You also try the UFO video game motion test, please?

**Comment by Endeavour18** [comment]
@IgorHoruzha No. The UFO is as smooth as the other accounts I tested. I think you're missing the point. Do you work for Valve?

**Comment by IgorHoruzha** [comment]
>Do you work for Valve?

I do not work for Valve, but I'm interested in such type of issues.

**Comment by IgorHoruzha** [comment]
>I think you're missing the point.

Could You express the point, please?

**Comment by Endeavour18** [comment]
@IgorHoruzha The point is that I'm lagging exclusively on my account, which means that the problem is not with my PC or my settings because I can play with any other account without that stutter.

**Comment by Endeavour18** [comment]
@IgorHoruzha It's not placebo. I can see the stutter even at 60fps, now imagine playing on my 144hz monitor. It's so annoying.

**Comment by IgorHoruzha** [comment]
> It's not placebo. I can see the stutter even at 60fps on my 2nd monitor, now imagine playing on my 144hz main monitor. It's so annoying.

I see Your point. Could You try to show it as clearly as possible, maybe by using a video editor to highlight the moment when the issue appears, please?

**Comment by IgorHoruzha** [comment]
Also, could You set the FPS to 30 in Dota 2 using the fps_max 30 command and check if the game looks the same as when the issue occurs?

**Comment by IgorHoruzha** [comment]
Also, could You try  fps_max 500, please?

**Comment by Endeavour18** [comment]
@IgorHoruzha All timestamps are on the issue and I already tried capping the fps.

**Comment by IgorHoruzha** [comment]
I downloaded them and checked at 0.1x speed. I found a 30fps frame rate and no stutters.

**Comment by IgorHoruzha** [comment]
>I downloaded them and checked at 0.1x speed. I found a 30fps frame rate and no stutters.

Maybe I made a mistake somewhere, so it would be helpful if You could highlight the moment.

**Comment by Endeavour18** [comment]
@IgorHoruzha you just have to watch them with the max resolution possible on youtube at 60fps and see for yourself. Look at [this](https://youtu.be/HA2CDn-uTu4?t=588) time stamp and watch carefuly. I DO NOT see 60 frames here (watching the video), either I see 100+ fps (playing).

**Comment by Endeavour18** [comment]
Feels like the entire gameplay is running at I don't know 40 FPS, even with the number showing 100+. That's the whole point.

**Comment by IgorHoruzha** [comment]
> @IgorHoruzha you just have to watch them with the max resolution possible on youtube at 60fps and see for yourself. Look at [this](https://youtu.be/HA2CDn-uTu4?t=588) time stamp and watch carefuly.

I tried watching carefully the fragment, but it’s still the same 30fps video with no stutters.

https://github.com/user-attachments/assets/63ff317a-31ac-43b8-a560-02a3679faf75

> DO NOT see 60 frames here (watching the video), neither I see 100+ fps (playing).
 
 Could You check the video frame rate in settings, please?
![image](https://github.com/user-attachments/assets/42c353b2-f891-43f6-a3dd-bb99fecd9570)

**Comment by IgorHoruzha** [comment]
>Feels like the entire gameplay is running at I don't know 40 FPS, even with the number showing 100+. That's the whole point.

Could You set the FPS to 40 using the console command fps_max 40 and compare the results, please?

**Comment by IgorHoruzha** [comment]
#21567 possibly related.

**Comment by Endeavour18** [comment]
@IgorHoruzha maybe you downloaded the video at 30 fps? Try watching directly from youtube, as I mentioned.

**Comment by IgorHoruzha** [comment]
I tried, but YouTube's slowest speed is 0.25x, and it doesn't allow me to view the video's frame rate or scroll by milliseconds. I didn’t notice any stutters either.  IMO It looks like YouTube vide is also 30fps frame rate. 

It's important to be able to watch the video frame by frame to detect any inconsistencies for sure.

**Comment by IgorHoruzha** [comment]
It doesn't meant that I'm going to say that I think that You do not have the issue. It means only that I have not seen it clearly yet. So, If You could highlight it, it would be very brilliant.

**Comment by Endeavour18** [comment]
You dont have to pause or slow it down. Just look at the hero spinning around and the image microstutter. I've shown this to some friends and they can see it too.  If you can't see that, this conversation is useless.

**Comment by IgorHoruzha** [comment]
>You dont have to pause or slow it down. Just look at the hero spinning around and the image microstutter. I've shown this to some friends and they can see it too. If you can't see that, this conversation is useless.


The way is usual to 30fps videos. IMO this way 30 fps looks

https://github.com/user-attachments/assets/76577bdb-fe7b-423e-9c77-58d3901b6f78

**Comment by Endeavour18** [comment]
@IgorHoruzha yeah you're correct. The microstutter that I see is more than that.

**Comment by IgorHoruzha** [comment]
>yeah you're correct. The microstutter that I see is more than that.

Could You try recording them using OBS Studio at a 60fps frame rate, please?
And if possible, use the same scenario without any stutters.

Do You experience the issue in Demo mode or locally hosted games?

**Comment by Endeavour18** [comment]
Microstuttering even more now after CC2 and 7.37d

**Comment by acttogreen** [comment]
hi, is there any fix about this issues. i've same setup and same issues. any idea ?

**Comment by IgorHoruzha** [comment]
Hello, was the issue resolved?

**Comment by Endeavour18** [comment]
@IgorHoruzha No.

**Comment by Endeavour18** [comment]
Turns out that the problem was solved. Had to buy a 9800X3D to play this 10 year old game. Thanks for the help.
